### PR TITLE
[Programmatic payments] list-stored-payment-methods webhook

### PR DESCRIPTION
**Pull request by korycins**
I want to merge this change because it adds a new webhook as described in: https://github.com/saleor/saleor/issues/13384

This webhook will be used to fetch all payment methods from payment providers. The implementation also included the cached flow, the same as added in the list shipping method. 
I've also made a small refactor of the logic related to the cache. Now it can be used out of the box - it is used in by external-shipping-methods webhook and the new list-shipping-methods.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/payment/__init__.py [hunk 1]
```diff
@@ -233,3 +233,20 @@ class TransactionEventType:
         (CANCEL_REQUEST, "Represents cancel request"),
         (INFO, "Represents an info event"),
     ]
+
+
+class TokenizedPaymentFlow:
+    """Represents possible tokenized payment flows that can be used to process payment.
+
+    Added in Saleor 3.15.
```
Comment (fowczarek) [inline saleor/payment/__init__.py:241]: We should use `ADDED_IN_315` here?

File: saleor/graphql/webhook/subscription_types.py [hunk 2]
```diff
@@ -1738,6 +1742,60 @@ class Meta:
         doc_category = DOC_CATEGORY_PAYMENTS
 
 
+class ListPaymentMethods(SubscriptionObjectType):
```
Comment (fowczarek) [inline saleor/graphql/webhook/subscription_types.py:1745]: Maybe? 
```suggestion
class ListStoredPaymentMethods(SubscriptionObjectType):
```

File: saleor/graphql/webhook/subscription_types.py [hunk 3]
```diff
@@ -1738,6 +1742,60 @@ class Meta:
...
+            "Channel in context which was used to fetch the list of payment methods"
+        ),
+        required=True,
+    )
+    value = graphene.Field(
+        Money,
+        description=(
+            "Amount that the payment method must support. If not provided by the user, "
+            "0 will be used as default value"
+        ),
```
Comment (IKarbowiak) [inline saleor/graphql/webhook/subscription_types.py:1765]: Lack of dots at the end of descriptions.

File: saleor/plugins/webhook/tests/test_list_payment_methods_webhook.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    assert response
+    assert response == get_list_payment_methods_from_response(
+        list_payment_methods_app, webhook_list_payment_methods_response
+    )
+
+
+@mock.patch("saleor.plugins.webhook.tasks.cache.set")
+@mock.patch("saleor.plugins.webhook.tasks.cache.get")
+@mock.patch("saleor.plugins.webhook.tasks.send_webhook_request_sync")
+def test_list_payment_methods_uses_cache_if_available(
```
Comment (IKarbowiak) [inline saleor/plugins/webhook/tests/test_list_payment_methods_webhook.py:165]: Do we have also tests for checking if cache is properly set?

**Review by IKarbowiak — APPROVED** [review]

File: saleor/plugins/webhook/tests/test_list_payment_methods_webhook.py (hunk 4 above)
Comment (korycins) [reply saleor/plugins/webhook/tests/test_list_payment_methods_webhook.py]: Yes, I am checking it in `test_list_stored_payment_methods_with_subscription_payload`, `test_list_stored_payment_methods_with_static_payload` both of the tests as a result of the action returns the fetched response from the App, and updates the cache.

File: saleor/plugins/webhook/utils.py [hunk 5]
```diff
@@ -289,3 +297,146 @@ def get_meta_code_key(app: App) -> str:
...
+    return PaymentMethodCreditCardInfo(
+        brand=str(brand),
+        last_digits=str(last_digits),
+        exp_year=exp_year,
+        exp_month=exp_month,
+        first_digits=str(first_digits) if first_digits else None,
+    )
+
+
+def get_payment_method_from_response(
```
Comment (maarcingebala) [inline saleor/plugins/webhook/utils.py:376]: Maybe it's time to extract payment-related utils to a separate file and not clutter `utils.py` anymore, there is everything now from general utils to functions specific to particular webhooks.

**Review by fowczarek — APPROVED** [review]

**Pull request merged**